sopel/modules/currency.py

```diff
@@ -2,109 +2,201 @@
+class CurrencySection(StaticSection):
+    fixer_io_key = ValidatedAttribute('fixer_io_key', default=None)
+    auto_convert = ValidatedAttribute('auto_convert', parse=bool, default=False)
 
 
-@commands('cur', 'currency', 'exchange')
-@example('.cur 20 EUR in USD')
-def exchange(bot, trigger):
+def configure(config):
+    config.define_section('currency', CurrencySection, validate=False)
+    config.currency.configure_setting('fixer_io_key', 'Optional API key for Fixer.io (leave blank to use exchangeratesapi.io):')
+    config.currency.configure_setting('auto_convert', 'Automatically respond to currency conversion lines?')
+
+
+def setup(bot):
+    bot.config.define_section('currency', CurrencySection)
+
+
+def update_rates(bot):
+    global rates_fiat_json, rates_btc_json, rates_updated
+
+    # If we have data that are less than 24h old, return
+    if time.time() - rates_updated < 24 * 60 * 60:
+        return
+
+    # Update crypto rates
+    response = requests.get(CRYPTO_URL)
+    response.raise_for_status()
+    rates_btc_json = response.json()
+
+    # Update fiat rates
+    if bot.config.currency.fixer_io_key is not None:
+        response = requests.get(FIXER_URL.format(bot.config.currency.fixer_io_key))
+        if not response.json()['success']:
+            LOGGER.error(str(response.json()['error']))
+            return bot.reply('Sorry, something went wrong')
+    else:
+        response = requests.get(FIAT_URL)
+
+    response.raise_for_status()
+    rates_fiat_json = response.json()
+    rates_updated = time.time()
+    rates_fiat_json['rates']['EUR'] = 1.0  # Put this here to make logic easier
+
+
+def btc_rate(code, reverse=False):
+    search = 'BTC{}'.format(code)
+
+    if search in rates_btc_json:
+        rate = rates_btc_json[search]['averages']['day']
+    else:
+        return UNSUPPORTED_CURRENCY.format(code)
+
+    if reverse:
+        return 1 / rate
+    else:
+        return rate
+
+
+def get_rate(of, to):
+    of = of.upper()
+    to = to.upper()
+
+    if of == 'BTC' or to == 'BTC':
+        if of == 'BTC':
+            code = to
+            reverse = False
+        else:
+            code = of
+            reverse = True
+
+        return btc_rate(code, reverse)
+
+    if of not in rates_fiat_json['rates']:
+        return UNSUPPORTED_CURRENCY.format(of)
+
+    if to not in rates_fiat_json['rates']:
+        return UNSUPPORTED_CURRENCY.format(to)
+
+    return (1 / rates_fiat_json['rates'][of]) * rates_fiat_json['rates'][to]
+
+
+def exchange(bot, match):
     """Show the exchange rate between two currencies"""
-    if not trigger.group(2):
-        return bot.reply("No search term. An example: .cur 20 EUR in USD")
-    match = regex.match(trigger.group(2))
     if not match:
-        # It's apologetic, because it's using Canadian data.
-        bot.reply("Sorry, I didn't understand the input.")
+        bot.reply(UNRECOGNIZED_INPUT)
+        return NOLIMIT
+
+    try:
```

Comment:
Catch the exception from `update_rates()` here, and display the message from there, here instead.